cmd/fluxd/main.go

```diff
@@ -142,90 +146,106 @@ func main() {
 	}
 
 	// Platform component.
-	var clusterVersion string
+
+	var fluxCluster cluster.Cluster
 	var sshKeyRing ssh.KeyRing
-	var k8s cluster.Cluster
+	var clusterVersion string
+	var manifests cluster.Manifests
 	var image_creds func() registry.ImageCreds
-	var k8sManifests cluster.Manifests
-	{
-		restClientConfig, err := rest.InClusterConfig()
-		if err != nil {
-			logger.Log("err", err)
-			os.Exit(1)
-		}
 
-		restClientConfig.QPS = 50.0
-		restClientConfig.Burst = 100
+	if *namespace == "" {
```

Comment:
This seems like a strange way to determine if Kubernetes is the intended target rather than Docker Swarm. I would prefer something more explicit, against which the arguments can be sanity-checked. That is, if you say you're using Swarm, it breaks when you supply arguments to do with Kubernetes, and vice-versa.